### PR TITLE
[Metricbeat] Fix fields not being parsed correctly in postgresql/database

### LINKED ISSUE #25301: [Metricbeat] [PostgreSQL] module [database] metricset - wrong mapping for blk_read_time and blk_write_time
- Version: metricbeat 7.11 + PostgreSQL 11
- Operating System: k8s v1.18.9+k3s1 under k3OS v0.11.1
- Discuss Forum URL: https://discuss.elastic.co/t/metricbeat-module-postgresql-wrong-mapping-of-blk-read-time-and-blk-write-time/270722
- Steps to Reproduce: 
1. in Metricbeat: enable PostgreSQL module
2. in PostgreSQL: enable `track_io_timing` via `postgresql.conf` --> uncomment or add `track_io_timing = on`
3. in Metricbeat: connect the Metricbeat instance with PostgreSQL module to your PostgreSQL database + enable `database` metricset
4. check that fields `postgresql.database.blocks.time.read.ms` and `postgresql.database.blocks.time.write.ms` are missing while expected (due to step n°2).

- Detailed description:

According to [beats/metricbeat/module/postgresql/database/data.go](https://github.com/elastic/beats/blob/master/metricbeat/module/postgresql/database/data.go#L40-L41), the following fields are mapped as Integer:
```
	    "blocks": s.Object{
...
		"time": s.Object{
			"read":  s.Object{"ms": c.Int("blk_read_time")},
			"write": s.Object{"ms": c.Int("blk_write_time")},
		},
...
```

But according to the official PostgreSQL documentation, theses columns are **double precision** - see [PostgreSQL: Documentation: 9.2: The Statistics Collector](https://www.postgresql.org/docs/9.2/monitoring-stats.html#PG-STAT-DATABASE-VIEW)

Due to this wrong mapping, once the "track_io_timing" feature is enabled at PostgreSQL side then the fields `postgresql.database.blocks.time.read.ms` and `postgresql.database.blocks.time.write.ms` are always missing (more than probably due to an error while mapping).

### PR DESCRIPTION
**Pull request by sayden**
## What does this PR do?
Fix `postgresql.database.blocks.time.read.ms` and `postgresql.database.blocks.time.write.ms` being captured and reported as a `long` instead of `double` thus not being reported at all.

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #25301

### CODE REVIEW

File: metricbeat/module/postgresql/database/_meta/fields.yml [hunk 1]
```diff
@@ -36,12 +36,12 @@
...
         buffer cache, not the operating system's file system cache).
     - name: blocks.time.read.ms
-      type: long
+      type: double
       description: >
         Time spent reading data file blocks by backends in this database, in
         milliseconds.
     - name: blocks.time.write.ms
-      type: long
+      type: double
```
Comment (jsoriano) [inline metricbeat/module/postgresql/database/_meta/fields.yml:44]: Please, double check if this change is problematic in a deployment with both versions of Metricbeat, we had recently some bad experiences with type changes that looked safe.

The kind of test you could do is:
* Install Metricbeat 7.15 and create a visualization using this field (even if it is zero).
* Double-check that you have a `metricbeat-*` index pattern where this field appears.
* In the same deployment, update to Metricbeat with this fix, and check if there is any problem with the visualization or with the index pattern.

Comment (jsoriano) [reply metricbeat/module/postgresql/database/_meta/fields.yml:44]: cc @kaiyan-sheng

**Comment by ruflin** [comment]
Should this also be backported to 7.16?

File: metricbeat/module/postgresql/database/_meta/fields.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply metricbeat/module/postgresql/database/_meta/fields.yml:44]: Yeah please double check here 😂 Not sure if it's the same case but I accidentally made a breaking change for changing field type from scaled float to long.

**Comment by jlind23** [comment]
@sayden what should we do with this PR?

**Comment by ruflin** [comment]
I removed the backport labels on this PR to make sure it doesn't get backported to any old branches by accident. We can add backport labels when this gets merged.

**Comment by ruflin** [comment]
For the PR itself, lets get this over the line as it is valuable fix.

**Comment by ishleenk17** [comment]
@kaiyan-sheng @jsoriano : 
This issue was brought up recenlty by a support engineer. 
Any reason why this was not merged ?
Looks like a customer is facing an issue due to this.

**Comment by jsoriano** [comment]
@ishleenk17 looking to the comment history only [some testing](https://github.com/elastic/beats/pull/29051#discussion_r753389963) and the decision to backport was pending, change looks good and valuable for the rest. Not sure why this was closed. Feel free to take it over and complete it.

**Comment by PBoff** [comment]
@jsoriano and @ishleenk17 - Do we have any timelines for this?.

**Comment by PBoff** [comment]
@jsoriano and @ishleenk17 - Do we have any timelines for this?.

**Comment by ishleenk17** [comment]
@kush-elastic : Could you please share the latest on this ?

**Comment by kush-elastic** [comment]
I have started working on this. I will post the results of my findings soon.
There are some scenarios I need to verify for post change as discussed with @ishleenk17.
Thanks

**Comment by kush-elastic** [comment]
@PBoff and @ishleenk17,
I have raised PR with changes and verified the scenarios presented before. Let's wait for the PR to be reviewed and merged.
Here's PR [Link](https://github.com/elastic/beats/pull/37720).

**Pull request closed without merging**